### ISSUE TITLE
Connection (sasl+login) 3 times at row on failed login

### ISSUE THREAD

**Issue by amerlyq** [neomutt/neomutt#766]
By the results of conversation with @gahr on IRC 2017-09-18.
See how imap_authenticate() is called 3 times instead of only one.
Look at function traces extracted for my case https://pastebin.com/aGD7idgJ
And corresponding logs https://gist.github.com/amerlyq/a32fab2a7a33d68ca715b1d5ab099aa2

Note: sasl + login is separate problem which results in 6 "LOGIN failed" answers from imap server.

**Comment by gahr** [comment]
To summarize: we have two separate issues here, your configuration seems to:

1. trigger both a SASL LOGIN and an IMAP LOGIN attempts
2. trigger three times `imap_open_mailbox`

I would first try to investigate why you're calling `imap_open_mailbox` three times. It might be because of `mutt_buffy_check` being called after some poll timeout or because of the status bar group `p` calling out to imap to get the number of postponed messages... or a combination..

Fixing this might be difficult, though, as it would imply keeping all interactive components frozen until the login has successfully been finalized.

**Comment by amerlyq** [comment]
In earlier attached func trace you can see, that `imap_open_mailbox()` is triggered only once, which triggers `imap_open_connection()` -- also only once at `imap_conn_find()`. 

But  there are 3x `imap_authenticate()`, each after `imap_conn_find()` (of total 14x).

1) inside `imap_open_mailbox()` on actual mailbox open

2) in index chain:

 mutt_index_menu
   mutt_buffy_check
   mutt_update_num_postponed
   imap_buffy_check
    imap_get_mailbox.constprop

3) on stats update:

 mutt_index_menu
  index_menu_redraw
   menu_status_line
    mutt_expando_format
     status_format_str
      mutt_num_postponed
       imap_status
        imap_get_mailbox.constprop

Actually, I couldn't even begin to comprehend why interface would query some mailbox data even before imap was successfully opened on login. And why there isn't any mutex to block until connection is done -- because auth is unshared resource, even if separate connections are shared.

**Comment by amerlyq** [comment]
Is there any user-side crotch e.g. disable "mutt_update_num_postponed" and "imap_status" in muttrc and manually toggle then once you successfully logged in ?
